base/compiler/typeinfer.jl

```diff
@@ -633,7 +633,12 @@ end
 
 
 function return_type(@nospecialize(f), @nospecialize(t))
-    params = Params(ccall(:jl_get_tls_world_age, UInt, ()))
+    world = ccall(:jl_get_tls_world_age, UInt, ())
+    return ccall(:jl_call_in_typeinf_world, Any, (Ptr{Ptr{Cvoid}}, Cint), Any[_return_type, f, t, world], 4)
```

Comment:
Ptr{Any} might be more accurate, although legacy C compatibility generally demands that Cvoid is the same, I think escape analysis may see them differently.